### PR TITLE
Movecraft-Combat (MCC)

### LINKED ISSUE #290: Craft.sink or CraftReleaseEvent.setSinking
**Is your feature request related to a problem? Please describe.**
You currently can not trigger a craft that is being released to sink rather than release.

**Describe the solution you'd like**
Either a method in CraftReleaseEvent that changes it from a release to a sink, or a method to sink the released craft.

### PR DESCRIPTION
**Pull request by TylerS1066**
### Describe in detail what your pull request acomplishes
This pull request moves many of the functions found in Movecraft to an addon plugin, found here: https://github.com/TylerS1066/MC-Combat

### Related issues:
- Fixes #290

### Checklist
- [ ] Proper internationalization
- [x] Compiled/tested

### CODE REVIEW

**Comment by TylerS1066** [comment]
I now believe that MCC v0.4.24 is sufficiently stable to remove this functionality from Movecraft itself.

**Comment by TylerS1066** [comment]
This PR has had some major changes since the last review, in addition to relying on code from #313 and #311 .  Please review (and merge) those before reviewing this PR.

**Comment by eirikh1996** [comment]
Changes seems good so far. Still, I will leave it up to @cccm5 to give a final decision to merge it or if further changes are needed

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java [hunk 1]
```diff
@@ -728,12 +723,11 @@ private int dropDistance (BitmapHitBox hitBox) {
...
                 //This has to be subtracted by one, or non-passthrough blocks will be within the y drop path
                 //obstructing the craft
-                MovecraftLocation dropped = translated.translate(0, dropDistance - 1 , 0);
-                Material testType = dropped.toBukkit(craft.getW()).getBlock().getType();
+                Material testType = translated.translate(0, dropDistance , 0).toBukkit(craft.getW()).getBlock().getType();
                 hitGround = testType != Material.AIR &&
                         !craft.getType().getPassthroughBlocks().contains(testType) &&
                         !(craft.getType().getHarvestBlocks().contains(testType) &&
-                        craft.getType().getHarvesterBladeBlocks().contains(ml.toBukkit(craft.getW()).getBlock().getType())) ||
+                        craft.getType().getHarvesterBladeBlocks().contains(ml.translate(0, 1, 0).toBukkit(craft.getW()).getBlock().getType())) ||
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java:730]: is this a bug with harvest blocks being fixed or an accidental reversion?

Comment (TylerS1066) [reply modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java:730]: It appears to be a mistake revision from https://github.com/APDevTeam/Movecraft/pull/292

I'll fix it and push.

Comment (eirikh1996) [reply modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java]: It was done to fix an issue where crafts affected by gravity would not drop to the ground

**Comment by TylerS1066** [comment]
I'm going to take a fine tooth comb through everything that renders as a diff and verify that it was either an intended change or incorrectly rendered due to a PR merge.

**Comment by TylerS1066** [comment]
Fixed all unintentional reversions.

**Pull request merged**